x/blob/README.md

```diff
@@ -2,44 +2,141 @@
+
+After a block has been created, the user can verify that their data was included
+in a block via a blob inclusion proof. A blob inclusion proof uses the
+`ShareCommitment` in the original transaction and subtree roots of the block's
+data square to prove to the user that the shares that compose their original
+data do in fact exist in a particular block.
 
 ## State
 
-The blob module doesn't maintain it's own state.
+The blob module doesn't maintain it's own state outside of a single param.
 
 When a `MsgPayForBlob` is processed, it consumes gas based on the blob size.
 
 ## Messages
 
-- [`MsgPayForBlob`](https://github.com/celestiaorg/celestia-app/blob/8b9c4c9d13fe0ccb6ea936cc26dee3f52b6f6129/proto/blob/tx.proto#L39-L44) pays for the blob to be included in the block.
-
-## PrepareProposal
+- [`MsgPayForBlobs`](https://github.com/celestiaorg/celestia-app/blob/v1.0.0-rc2/proto/celestia/blob/v1/tx.proto#L16-L31)
+  pays for a set of blobs to be included in the block. Transactions containing
+  this `sdk.Msg` are better known as "PFBs".
+
+```proto
+// MsgPayForBlobs pays for the inclusion of a blob in the block.
+message MsgPayForBlobs {
+  string signer = 1;
+  // namespaces is a list of namespaces that the blobs are associated with. A
+  // namespace is a byte slice of length 33 where the first byte is the
+  // namespaceVersion and the subsequent 32 bytes are the namespaceId.
+  repeated bytes namespaces = 2;
+  repeated uint32 blob_sizes = 3;
+  // share_commitments is a list of share commitments (one per blob).
+  repeated bytes share_commitments = 4;
+  // share_versions are the versions of the share format that the blobs
+  // associated with this message should use when included in a block. The
+  // share_versions specified must match the share_versions used to generate the
+  // share_commitment in this message.
+  repeated uint32 share_versions = 8;
+}
+```
 
-When a block producer is preparing a block, they must perform an extra step for `BlobTx`s so that end-users can find the blob shares relevant to their submitted `BlobTx`. In particular, block proposers wrap the `BlobTx` in the PayForBlobs namespace with the index of the first share of the blob in the data square. See [Non-interactive Default Rules](https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md#non-interactive-default-rules) for more details.
+### Generating the `ShareCommitment`
 
-Since `BlobTx`s can contain multiple blobs, the `BlobTx` is wrapped with one share index per blob in the transaction. The index wrapped transaction is called an [IndexWrapper](https://github.com/celestiaorg/celestia-core/blob/2d2a65f59eabf1993804168414b86d758f30c383/proto/tendermint/types/types.proto#L192-L198) and this is the type that gets marshalled and written to the PayForBlobNamespace.
+1. Split the blob into shares of size `appconsts.ShareSize`
```

Comment:
A link to the `appconsts.ShareSize` would be great. 
Also, if this constant is  covered in other specs, a link to that specs would be even better (instead of/alongside  `appconsts.ShareSize` where  `appconsts` refers to a go package).